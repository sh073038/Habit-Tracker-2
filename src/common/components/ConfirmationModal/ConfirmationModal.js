// @flow

import * as React from 'react'
import Button from 'material-ui/Button'
import Dialog, {
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from 'material-ui/Dialog'

type Props = {
  open: boolean,
  onClose: () => void,
  onConfirm: () => void,
  title: ?string,
  children: React.Node,
}

const RemoveGoal = ({ open, onClose, onConfirm, title, children }: Props) => (
  <Dialog
    open={open}
    onClose={onClose}
    aria-labelledby="alert-dialog-title"
    aria-describedby="alert-dialog-description"
  >
    <DialogTitle id="alert-dialog-title">{title || 'Confirm'}</DialogTitle>
    <DialogContent>
      <DialogContentText id="alert-dialog-description">{children}</DialogContentText>
    </DialogContent>
    <DialogActions>
      <Button onClick={onConfirm} color="primary" autoFocus>
        Confirm
      </Button>
      <Button onClick={onClose}>Cancel</Button>
    </DialogActions>
  </Dialog>
)

export default RemoveGoal
