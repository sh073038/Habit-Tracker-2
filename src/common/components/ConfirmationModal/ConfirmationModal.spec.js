import React from 'react'
import { shallow } from 'enzyme'
import { DialogTitle } from 'material-ui/Dialog'

import ConfirmationModal from './ConfirmationModal'

describe('ConfirmationModal', () => {
  it('renders component', () => {
    const wrapper = shallow(
      <ConfirmationModal title="Title" open onClose={jest.fn()} onConfirm={jest.fn()} />,
    )
    expect(wrapper.getElement()).toMatchSnapshot()
    expect(wrapper.props()).toMatchSnapshot()
    expect(wrapper.find(DialogTitle).props().children).toEqual('Title')
  })

  it('default title', () => {
    const wrapper = shallow(
      <ConfirmationModal title="Title" open onClose={jest.fn()} onConfirm={jest.fn()} />,
    )
    expect(wrapper.find(DialogTitle).props().children).toEqual('Title')
    const wrapper2 = shallow(<ConfirmationModal open onClose={jest.fn()} onConfirm={jest.fn()} />)
    expect(wrapper2.find(DialogTitle).props().children).toEqual('Confirm')
  })
})
