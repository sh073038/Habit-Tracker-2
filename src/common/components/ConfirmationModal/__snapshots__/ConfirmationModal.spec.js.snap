// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal renders component 1`] = `
<WithStyles(Dialog)
  aria-describedby="alert-dialog-description"
  aria-labelledby="alert-dialog-title"
  onClose={[MockFunction]}
  open={true}
>
  <WithStyles(DialogTitle)
    id="alert-dialog-title"
  >
    Title
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <WithStyles(DialogContentText)
      id="alert-dialog-description"
    />
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      autoFocus={true}
      color="primary"
      onClick={[MockFunction]}
    >
      Confirm
    </WithStyles(Button)>
    <WithStyles(Button)
      onClick={[MockFunction]}
    >
      Cancel
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;

exports[`ConfirmationModal renders component 2`] = `
Object {
  "aria-describedby": "alert-dialog-description",
  "aria-labelledby": "alert-dialog-title",
  "children": Array [
    <WithStyles(DialogTitle)
      id="alert-dialog-title"
    >
      Title
    </WithStyles(DialogTitle)>,
    <WithStyles(DialogContent)>
      <WithStyles(DialogContentText)
        id="alert-dialog-description"
      />
    </WithStyles(DialogContent)>,
    <WithStyles(DialogActions)>
      <WithStyles(Button)
        autoFocus={true}
        color="primary"
        onClick={[MockFunction]}
      >
        Confirm
      </WithStyles(Button)>
      <WithStyles(Button)
        onClick={[MockFunction]}
      >
        Cancel
      </WithStyles(Button)>
    </WithStyles(DialogActions)>,
  ],
  "onClose": [MockFunction],
  "open": true,
}
`;
